src/sys/event.rs

```diff
@@ -274,10 +263,13 @@ type type_of_nchanges = c_int;
 #[cfg(target_os = "netbsd")]
 type type_of_nchanges = size_t;
 
-pub fn kevent_ts(kq: RawFd,
-              changelist: &[KEvent],
-              eventlist: &mut [KEvent],
-              timeout_opt: Option<timespec>) -> Result<usize> {
+pub fn kevent<T: Into<TimeSpec>>(kq: RawFd, changelist: &[KEvent], eventlist: &mut [KEvent], timeout: Option<T>) -> Result<usize> {
```

Comment:
Instead `pub fn kevent<T: Into<Option<TimeSpec>>>(kq: RawFd, changelist: &[KEvent], eventlist: &mut [KEvent], timeout: T) -> Result<usize>`

Reply:
I don't think that is possible, since you'll get trait bound issue such as: `'std::convert::From<std::time::Duration>' is not implemented for 'std::option::Option<nix::sys::time::TimeSpec>'`